newsfragments/2534.bugfix.rst

```diff
@@ -0,0 +1 @@
+`trio.from_thread.run` and `trio.from_thread.run_sync` no longer raise `RuntimeError` when called in a deadlock-free way from a thread running Trio.
```

Comment:
instead of "from a thread running Trio", maybe elaborate "to enter one thread running Trio from another"? I also wonder if it would be good to add a disclaimer here that using them this way is still suboptimal (will block the originating thread's event loop) but we support it because it can be useful in some cases.